src/privileged_extensions.c

```diff
@@ -173,6 +173,19 @@ void handle_create_extension(
         }
     }
 
+    // Handle pg_cron 1.4 -> 1.5 breaking change.
+    if (strcmp(stmt->extname, "pg_cron") == 0) {
+        ListCell *option_cell = NULL;
+
+        foreach (option_cell, stmt->options) {
+            DefElem *defel = (DefElem *)lfirst(option_cell);
+
+            if (strcmp(defel->defname, "schema") == 0) {
+                defel->arg = (Node *)makeString("pg_catalog");
+            }
+        }
+    }
```

Comment:
@soedirgo This is too ad-hoc. I get that this is a solution for the Supabase platform but why make an extension OSS if it's not generally useful? 

(I think the correct way to solve this would be to fork `pg_cron` but I get that this is too complicated for supabase infra right now. Hopefully this will be easier with `nix-postgres`)

So how about making this configurable instead? Maybe like:

```
supautils.extensions_forced_schema = '{"pg_cron": "pg_catalog"}';
```

Reply:
Hm, this could also be part of constrained extensions:

```
supautils.constrained_extensions = '{"pg_cron": {"schema": "pg_catalog", "cpu": 16, "mem": "1 GB", "disk": "500 MB"}}'
```

But it would make sense to keep this functionality separate.

@soedirgo If you'd like I can help you with the json parsing part. You could define a struct like:

https://github.com/supabase/supautils/blob/bf9c1341ba059036f90733f9d868187867c9d30f/src/constrained_extensions.h#L6-L12

And then use a harcoded array on `supautils.c`:

https://github.com/supabase/supautils/blob/bf9c1341ba059036f90733f9d868187867c9d30f/src/supautils.c#L66

Then I'll plug the json parsing.

Reply:
Yeah I don't love the solution either, but pg_cron is the only extension where the issue happened and it's the only extension (that I know of) that ties its permissions to the owner of the extension object. It doesn't make sense to make this more generic if we don't foresee this happening for other extensions.

If we want to make this configurable, I'd probably make this a boolean (`supautils.force_pg_cron_on_pg_catalog`)

Reply:
My initial solution was to run `drop extension pg_cron; create extension pg_cron schema pg_catalog;` in the `after-create.sql`, but that doesn't work because the `create extension` itself would fail (because it'd target the `public`/`extensions` schema) before the `after-create.sql` would run.

Reply:
Hmm I think I'll make this something like `supautils.extensions_parameter_overrides` - aside for `extschema`, we can extend this to `extversion` or even `extowner` if need be. That way it won't be too ad-hoc and is more likely to be useful for other extensions.

Reply:
> supautils.extensions_parameter_overrides - aside for extschema, we can extend this to extversion or even extowner if need be

@soedirgo That sounds really useful. Something that https://github.com/dimitri/pgextwlist doesn't have.